docs/rules/prefer-destructuring.md

```diff
@@ -33,7 +33,9 @@ This rule takes two sets of configuration objects; the first controls the types
 
 The first has two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently.  By default, both are `true`.
 
-The second has a single property, `enforceForRenamedProperties`, that controls whether or not the `object` destructuring rules are applied in cases where the variable requires the property being access to be renamed.
+The first has two properties, `VariableDeclarator` and `AssignmentExpression`, which can be used to turn on or off the destructuring requirement for each of those types independently. Each property accepts an object that accpets two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently for variable declarations and assignment expressions.  By default, `array` and `object` are set to true for both `VariableDeclarator` and `AssignmentExpression`.
```

Comment:
Sorry if this was unclear before, but I think we should continue to allow the `array` and `object` top-level options. Making breaking changes to rule schemas is generally a pretty big deal -- in addition to breaking users, it also tends to make shareable configs invalid, which makes upgrading very tedious -- so we try to almost never do it. (Changing the default behavior of a rule schema while keeping all previous schemas valid is a bit better, but that's a separate topic.)

To avoid extra complexity, we can disallow mixing the old and new behaviors -- for example, this would still be an illegal config:

```yml
rules:
  prefer-destructuring: [error, {array: true, VariableDeclarator: {array: false}}]
```